test/apps/iodemo/io_demo.cc

```diff
@@ -797,8 +797,9 @@ class DemoClient : public P2pDemoCommon {
 
     void close_uncompleted_servers(const char *reason) {
         for (size_t i = 0; i < _active_servers.size(); ++i) {
-            if (get_num_uncompleted(i) > 0) {
-                terminate_connection(_server_info[i].conn, reason);
+            if (get_num_uncompleted(_active_servers[i]) > 0) {
+                terminate_connection(_server_info[_active_servers[i]].conn,
```

Comment:
ahhhh .. array is changed during termination..
maybe need to copy active_servers to temp array?

Reply:
`i < array.size()` should work since we don't handle > 1 elem per iteration, also we can do iterating in reverse order to avoid skipping swapped element. IMO temp array will add complexity to keep indexes in valid state.

Reply:
maybe like this:
while (!empty) {
 index = _active_servers.back()
 terminate(index) /* terminate removes the entry from active_servers */
}

Reply:
but we don't want to terminate all active servers, only with uncompleted ops

Reply:
right..
so IMO better to create temp list of which indexes to remove, to avoid any assumptions about order and swap element

Reply:
ok